mlir/lib/Transforms/Utils/DialectConversion.cpp

```diff
@@ -1281,22 +1281,17 @@ Block *ConversionPatternRewriterImpl::applySignatureConversion(
     ConversionPatternRewriter &rewriter, Block *block,
     const TypeConverter *converter,
     TypeConverter::SignatureConversion &signatureConversion) {
-  MLIRContext *ctx = rewriter.getContext();
+  OpBuilder::InsertionGuard g(rewriter);
```

Comment:
note: Insertion guard is needed because `OpBuilder::createBlock` changes the insertion point.